frontend/src/js/components/devices/DeviceGroups.tsx

```diff
@@ -246,6 +246,11 @@ export const DeviceGroups = () => {
 
   const toggleMakeGatewayClick = () => setShowMakeGateway(toggle);
 
+  const changeLocation = (newLocation: string) => {
+    isInitialized.current = false;
+    setLocationParams({ pageState: { ...deviceListState, state: newLocation }, filters, selectedGroup });
+  };
```

Comment:
This I assume will lead to the `onDeviceStateSelectionChange` callback being reset on every render of `DeviceGroups` (iirc the `useCallback` is needed there to have a properly tracked `refreshTrigger` value). Please wrap the `changeLocation` in a `useCallback` too.